### PR TITLE
feat: Add Utils file and move common function to Utils

### PR DESCRIPTION
**Pull request by dreadpool2**
*moved showProgressBar functions to Utils 
*moved showCustomtab functions to Utils

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        CustomTabsIntent.Builder()
+                .setToolbarColor(ContextCompat.getColor(context, R.color.colorPrimaryDark))
+                .setCloseButtonIcon(BitmapFactory.decodeResource(context.resources, R.drawable.ic_arrow_back_white_cct_24dp))
+                .setStartAnimations(context, R.anim.slide_in_right, R.anim.slide_out_left)
+                .setExitAnimations(context, R.anim.slide_in_left, R.anim.slide_out_right)
+                .build()
+                .launchUrl(context, Uri.parse(URL))
+    }
+
+    fun showProgressBar(rootView: View, show: Boolean) {
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Did you run the app locally ? This function is not same everywhere

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: It won't work in an activity. I didn't include this in any activity files

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Not good, pass progressBar, not view

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import android.net.Uri
+import android.support.customtabs.CustomTabsIntent
+import android.support.v4.content.ContextCompat
+import android.view.View
+import kotlinx.android.synthetic.main.fragment_profile.view.*
+import org.fossasia.openevent.general.R
+
+object Utils {
+
+    fun setUpCustomTab(context: Context, url: String) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Name should be openUrl

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 3]
```diff
@@ -53,8 +54,4 @@ class LoginFragment : Fragment() {
         activity?.finish()
     }
 
-    private fun showProgress(enabled: Boolean) {
-        loginButton.isEnabled = !enabled
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: what about this

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Updated PR :+1:

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import android.support.v4.content.ContextCompat
+import android.view.View
+import android.widget.ProgressBar
+import kotlinx.android.synthetic.main.fragment_profile.view.*
+import org.fossasia.openevent.general.R
+
+object Utils {
+
+    fun openUrl(context: Context, url: String) {
+        var URL = url
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Follow the style guideline. URL should be lower case

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+                .setToolbarColor(ContextCompat.getColor(context, R.color.colorPrimaryDark))
+                .setCloseButtonIcon(BitmapFactory.decodeResource(context.resources, R.drawable.ic_arrow_back_white_cct_24dp))
+                .setStartAnimations(context, R.anim.slide_in_right, R.anim.slide_out_left)
+                .setExitAnimations(context, R.anim.slide_in_left, R.anim.slide_out_right)
+                .build()
+                .launchUrl(context, Uri.parse(URL))
+    }
+
+    fun showProgressBar(progressBar: ProgressBar, show: Boolean) {
+        progressBar.isIndeterminate = show
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: I don' think this step is needed. Confirm

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Updated the PR :

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 6]
```diff
@@ -12,10 +12,10 @@ import org.fossasia.openevent.general.R
 
 object Utils {
 
-    fun openUrl(context: Context, url: String) {
-        var URL = url
+    fun openUrl(context: Context, link: String) {
+        var url = link
         if (!url.startsWith("http://") && !url.startsWith("https://")) {
-            URL = "http://$url"
+            url = "http://$url"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Default to https

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Changed it

**Pull request merged**